Services/CO.CDP.Organisation.WebApi/Api/Organisation.cs

```diff
@@ -645,6 +645,85 @@ await useCase.Execute((organisationId, personInviteId))
 
         return app;
     }
+
+    public static RouteGroupBuilder UseManageApiKeyEndpoints(this RouteGroupBuilder app)
+    {
+        app.MapGet("/{organisationId}/api-keys",
+            async (Guid organisationId, IUseCase<Guid, IEnumerable<Model.AuthenticationKey>> useCase) =>
+               await useCase.Execute(organisationId)
+                   .AndThen(entities => entities != null ? Results.Ok(entities) : Results.NotFound()))
+           .Produces<List<Model.AuthenticationKey>>(StatusCodes.Status200OK, "application/json")
+           .Produces<ProblemDetails>(StatusCodes.Status401Unauthorized)
+           .Produces<ProblemDetails>(StatusCodes.Status404NotFound)
+           .ProducesProblem(StatusCodes.Status422UnprocessableEntity)
+           .Produces<ProblemDetails>(StatusCodes.Status500InternalServerError)
+           .WithOpenApi(operation =>
+           {
+               operation.OperationId = "GetAuthenticationKeys";
+               operation.Description = "Get authentication keys by Organisation ID.";
+               operation.Summary = "Get authentication keys information by Organisation ID.";
+               operation.Responses["200"].Description = "Authentication keys details.";
+               operation.Responses["401"].Description = "Valid authentication credentials are missing in the request.";
+               operation.Responses["404"].Description = "authentication keys information not found.";
+               operation.Responses["422"].Description = "Unprocessable entity.";
+               operation.Responses["500"].Description = "Internal server error.";
+               return operation;
+           });
+
+        app.MapPost("/{organisationId}/api-keys",
+            async (Guid organisationId, RegisterAuthenticationKey registerAuthenticationKey,
+                IUseCase<(Guid, RegisterAuthenticationKey), bool> useCase) =>
+
+                await useCase.Execute((organisationId, registerAuthenticationKey))
+                    .AndThen(_ => Results.Ok())
```

Comment:
When a new resource is created, and we do not want to return the response body, we should use `201 Created` with a `Location` header pointing to the created resource.

In your case, I do not think we have an endpoint to return a single API key, so we cannot return the `Location` header, but I think we should still return 201 created rather than 200.

Reply:
Done